examples/thread-duell/main.c

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (C) 2020 TUBA Freiberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <thread.h>
+#include <sched.h>
+
+#include <math.h>
+
+#include "xtimer.h"
+
+#include "sched_round_robin.h"
+
+#define PRINT_STEPS 10
+#define WORK_SCALE  1000
+
+void bad_wait(uint64_t us)
+{
+    /*keep the cpu busy waiting for some time to pass simulate working*/
+    uint64_t until = xtimer_now_usec64() + us;
+
+    while (xtimer_now_usec64() < until) {
```

Comment:
Can we rewrite this so it doesn't depend on xtimer64?

Something like this (untested):
```
void xtimer_busywait(int32_t us) {
 if (us < 0) { return};
 uint32_t target = xtimer_now() + us;
 while ((int32_t)(target - xtimer_now()) > 0) {}
}
```

Reply:
of cause, i would even go with xtimer_spin but that one is private to xtimer